Classes/core/entity/entity.cpp

```diff
@@ -36,14 +39,22 @@ void Enemy::on_tick(GridRef g) {
             auto ny = route::ssize(g.column) + dy;
             assert(nx >= 0 && nx < g.map.shape.height_ && ny >= 0 && ny < g.map.shape.width_);
             g.map.move_enemy_to(this->id, nx, ny);
-            g.on_enemy_move(*this, std::make_pair(route::ssize(g.row), route::ssize(g.column)), std::make_pair(nx, ny));
+            if(dx < -1||dx > 1||dy < -1||dy >1){
```

Comment:
Why add an `on_transport`?

Reply:
animation of transport is different from move

Reply:
deleted

Reply:
> animation of transport is different from move

Then it's ok to add it. Though I perfer providing a function (of the two coordinates) to tell whether it is transport or regular move or adding a `bool is_transport` in callback.